corehq/messaging/scheduling/forms.py

```diff
@@ -1407,3 +1454,10 @@ def allow_case_property_filter(self):
     @property
     def allow_date_case_property_filter(self):
         return False
+
+    def __init__(self, *args, **kwargs):
+        super(ConditionalAlertCriteriaForm, self).__init__(*args, **kwargs)
+        if self.initial.get('case_type'):
```

Comment:
Hmm, on further thought I'm not sure what's going on here.

Reply:
the check that's being done here is just to see if we are editing or not -if we're editing, then we shouldn't be allowed to change the case type because it makes it much faster to run the rule (and changing the case type is not a common use case anyway).

so what this line is doing is just checking if there is any value for case type in the form's initial, which gets computed in the superclass, `CaseRuleCriteriaForm`. because the only way there is a value for the initial case type is if we are editing.

Reply:
added a comment: https://github.com/dimagi/commcare-hq/pull/18840/commits/e422c22a0f120f8adbe8660590d6f54128e7df30